server/src/main/java/org/cloudfoundry/identity/uaa/oauth/UaaTokenServices.java

```diff
@@ -282,6 +283,14 @@ public OAuth2AccessToken refreshAccessToken(String refreshTokenValue, TokenReque
                 generateUniqueTokenId()
         );
 
+        if (authenticationData.clientAuth != null && CLIENT_AUTH_NONE.equals(authenticationData.clientAuth)) {
```

Comment:
This is already a very long method. Could you add the new code to a new method that's called from here?